lib/ClangImporter/ClangDerivedConformances.cpp

```diff
@@ -865,6 +865,45 @@ void swift::conformToCxxSequenceIfNeeded(
   }
 }
 
+void swift::conformToCxxStackIfNeeded(
```

Comment:
I think it would be good with some negative test cases for inferring conformance, just to make sure it gets handled properly. In particular I'm interested in how element types that are `~Escapable` and/or `~Copyable` are handled. Do they get a diagnostic, or is the conformance just omitted? Also, can `std::stack` be declared for forward declared types in C++? If so, it'd be good to have a test case for that too.

Reply:
Thanks for the suggestions!  
Could you provide an example to illustrate how to test types that are ~Escapable and/or ~Copyable, as well as forward declared types?  This would help me better understand your requirements and write the relevant test cases.

Reply:
Something like this:
```
struct SWIFT_NONESCAPABLE ne_t {
  int dummy;
};
void foo(std::stack<ne_t> s);

struct SWIFT_NONCOPYABLE nc_t {
  int dummy;
};
void bar(std::stack<nc_t> s);

struct fd_t;
void baz(std::stack<fd_t> s);
```

What kind of errors does importing a header like that, and trying to call those functions, result in?

Reply:
Got it. Thank you for your suggestion.

Reply:
I _think_ what would happen in those cases is that the `public extension CxxStack { ... }` would not work, because `~Copyable` and `~Escapable` constraints need to be explicitly opted out of (though I'm not familiar enough with that feature to say for sure yet). So the protocol conformance would either be unusable by the user (what I would expect to happen) or cause the compiler to crash/misbehave (what should not happen).

@CrazyFanFan what happens when you try to import those tests that @hnrklssn suggested?